arc/simulation/modal_orchestrator.py

```
@@ -143,9 +169,51 @@ async def execute_with_persistence(
         )
 
+        # Batch processing context
+        batch_processor = None
+        if db_api and simulation_id:
+            def on_batch_complete(batch_size: int, processing_time: float):
+                """Callback for batch completion events."""
+                records_per_sec = batch_size / processing_time if processing_time > 0 else 0
+                self.console.print(f"[dim]Batch processed: {batch_size} records in {processing_time:.2f}s ({records_per_sec:.1f} rec/s)[/dim]")
+
+            async with batch_processor_context(
+                db_api, 
```

Ruff (0.11.9)

Trailing whitespace

Remove trailing whitespace

(W291)